components/brave_new_tab_ui/async/today.ts

```diff
@@ -140,4 +140,38 @@ handler.on(Actions.anotherPageNeeded.getType(), async function (store) {
   store.dispatch(Actions.checkForUpdate())
 })
 
+handler.on<Actions.VisitDisplayAdPayload>(Actions.visitDisplayAd.getType(), async function (store, payload) {
+  const state = store.getState() as ApplicationState
+  const todayPageIndex = state.today.currentPageIndex
+  chrome.send('todayOnDisplayAdVisit', [
+    payload.ad.uuid,
+    payload.ad.creativeInstanceId
+  ])
+  const destinationUrl = payload.ad.targetUrl
+  if (!payload.openInNewTab) {
+    // Remember display ad location so we can scroll to it on "back" navigation
+    // We remember position and not ad ID since it can be a different ad on
+    // a new page load.
+    // TODO(petemill): Type this history.state data and put in an API module
+    // (see `reducers/today`).
+    storeInHistoryState({
+      todayAdPosition: todayPageIndex,
+      todayPageIndex,
+      todayCardsVisited: state.today.cardsVisited
+    })
+    // visit article url
+    // @ts-ignore
+    window.location = destinationUrl
+  } else {
+    window.open(destinationUrl, '_blank')
```

Comment:
Please also set `noopener` and `noreferrer`.

Reply:
I'm not so sure we need this, or if we do then we should update all the other links from the NTP.
- "_blank" implies noopener from chromium 88, on any website (and was the case for WebUI pages already)
- a WebUI URL chrome://xyz will never be used for the referrer

Reply:
Since `noopener` is now the default for these links and that Pete has verified the referrer doesn't leak out, then it's okay to leave `noreferrer` out.